packages-ts/starknet-gauntlet/src/dependencies/index.ts

```diff
@@ -6,15 +6,16 @@ export interface Env {
   providerUrl: string
   pk?: string
   account?: string
-  [key: string]: string // Custom env
+  withLedger?: boolean
+  [key: string]: any // Custom env
 }
 
 export interface Dependencies {
   logger: typeof logger
   prompt: typeof prompt
   makeProvider: (url: string) => IStarknetProvider
-  makeWallet: (pk: string, account?: string) => IStarknetWallet
-  makeEnv: (flags: Record<string, string>) => Env
+  makeWallet: (withLedger: boolean, pk?: string, account?: string) => Promise<IStarknetWallet>
+  makeEnv: (flags: Record<string, any>) => Env
```

Comment:
Let's try to avoid `any`
```suggestion
  makeEnv: (flags: Record<string, string | boolean>) => Env
```